version/src/lib.rs

```diff
@@ -83,21 +76,40 @@ impl fmt::Debug for Version {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(
             f,
-            "{}.{}.{} (src:{}; feat:{})",
+            "{}.{}.{} (src:{:08x}; feat:{}, client:{:?})",
             self.major,
             self.minor,
             self.patch,
-            match self.commit {
-                None => "devbuild".to_string(),
-                Some(commit) => format!("{commit:08x}"),
-            },
+            self.commit,
             self.feature_set,
+            self.client(),
         )
     }
 }
 
 impl Sanitize for Version {}
 
+impl From<u16> for ClientId {
+    fn from(client: u16) -> Self {
+        match client {
+            0u16 => Self::SolanaLabs,
+            _ => Self::Unknown(client),
+        }
+    }
+}
+
+impl TryFrom<ClientId> for u16 {
+    type Error = String;
+
+    fn try_from(client: ClientId) -> Result<Self, Self::Error> {
+        match client {
+            ClientId::SolanaLabs => Ok(0u16),
+            ClientId::Unknown(client @ 0u16..=0u16) => Err(format!("Invalid client: {client}")),
+            ClientId::Unknown(client) => Ok(client),
```

Comment:
sorry I'm being dumb. `@ 0u16..=0u16)` covers the entire `u16` range right, so under what conditions would this match be hit?

Reply:
This is now `client @ 0u16..=2u16` since I added 2 other client-ids (jito, firedancer).
Effectively it says that `client` in `ClientId::Unknown(client)` should not be 0, 1, or 2; because those are already mapped to known clients.
For example, `ClientId::Unknown(0)` is logically wrong because `0 => SolanaLabs`, so it is not an `Unknown` client.

Reply:
ah, right. Sorry foggy glasses today. I misread `0u16..=0u16` as something else :)